libvirt/data_source_libvirt_network.go

```diff
@@ -156,3 +156,76 @@ func resourceLibvirtNetworkDNSSRVRead(d *schema.ResourceData, meta interface{})
 
 	return nil
 }
+
+//// DHCP
+
+// a libvirt network DHCP host template datasource
+//
+// Datasource example:
+//
+// data "libvirt_network_dhcp_host_template" "k8smasters" {
+//   count = "${var.master_count}"
+//   ip = "${var.master_ips[count.index]}"
+//   mac = "${var.master_mac[count.index]}"
+//   name = "master-${count.index}"
+// }
+//
+// resource "libvirt_network" "k8snet" {
+//   ...
+//   dhcp = [{
+//     hosts = [ "${flatten(data.libvirt_network_dhcp_host_template.k8smasters.*.rendered)}" ]
+//   }]
+//   ...
+// }
+//
+func datasourceLibvirtNetworkDHCPHostTemplate() *schema.Resource {
+	return &schema.Resource{
+		Read: resourceLibvirtNetworkDHCPHostRead,
+		Schema: map[string]*schema.Schema{
+			"ip": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"mac": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Is this really `Required`? I guess users will use `ip` with `mac`/`name`/`mac`-and-`name`,  so I gues both `mac` and `name` should be `Optional: true`...

Reply:
In case of standalone dhcp datasource, mac is indeed a required one, as is the ip.